### PR TITLE
GRW-1033/ Fix / Small CPAS bugs

### PR DESCRIPTION
**Pull request by amandaflod**
## What?

Fixes for these three tiny bugs 🐞 

- Page doesn’t scroll to top after navigation → when navigating from checkout details to checkout payment on mobile IOS. 

- Button text is on two rows 

 - Discount label is cut off on mobile (Zak said ok to truncate text)

## Why?

**Ticket(s): [GRW-1033]**

[GRW-1033]: https://hedvig.atlassian.net/browse/GRW-1033?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 1]
```diff
@@ -32,6 +32,10 @@ export const CheckoutDetails = () => {
 
   const { quoteCartId } = useQuoteCartIdFromUrl()
 
+  useEffect(() => {
```
Comment (simonauner) [inline src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:35]: Perhaps a custom hook `useScrollToTop`?

File: src/client/pages/Checkout/shared/Footer.tsx [hunk 2]
```diff
@@ -47,6 +48,32 @@ const InnerWrapper = styled.div`
...
+  ${LinkButton} {
+    padding: 0.75rem 1rem;
+
+    ${MEDIUM_SMALL_SCREEN_MEDIA_QUERY} {
+      padding: 0.75rem 2rem;
+    }
+  }
+
+  ${MEDIUM_SMALL_SCREEN_MEDIA_QUERY} {
+    ${Button} {
```
Comment (simonauner) [inline src/client/pages/Checkout/shared/Footer.tsx:69]: Isn't this the same as `${Button} ${MEDIUM_SMALL_SCREEN_MEDIA_QUERY}` above?

File: src/client/pages/Checkout/shared/Footer.tsx [hunk 3]
```diff
@@ -47,6 +48,32 @@ const InnerWrapper = styled.div`
...
+      padding: 0.75rem 2rem;
+    }
+
+    ${LinkButton} {
```
Comment (simonauner) [inline src/client/pages/Checkout/shared/Footer.tsx:73]: And this is the same as `${LinkButton} ${MEDIUM_SMALL_SCREEN_MEDIA_QUERY}` above?

File: src/client/pages/Checkout/shared/Footer.tsx (hunk 2 above)
Comment (amandaflod) [reply src/client/pages/Checkout/shared/Footer.tsx]: Yes, I must have left that in by mistake! 😰

Comment (simonauner) [reply src/client/pages/Checkout/shared/Footer.tsx]: It happens 😃

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx (hunk 1 above)
Comment (robinandeer) [reply src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:35]: This is what React Router docs recommend:

https://v5.reactrouter.com/web/guides/scroll-restoration/scroll-to-top

**Review by robinandeer — APPROVED** [review]
Nice! Merge or try the custom hook :D

**Review by simonauner — APPROVED** [review]

**Pull request merged**